x/delegation/keeper/share.go

```diff
@@ -0,0 +1,210 @@
+package keeper
+
+import (
+	sdkmath "cosmossdk.io/math"
+	assetstype "github.com/ExocoreNetwork/exocore/x/assets/types"
+	delegationtypes "github.com/ExocoreNetwork/exocore/x/delegation/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// TokensFromShares calculate the token amount of provided shares, then truncated to Int
+// It uses `LegacyDec.Quo` to calculate the quotient, `LegacyDec.Quo` perform a bankers
+// rounding quotient, so the calculated token amount may be either larger or smaller
+// due to precision rounding issues. But it's acceptable, because bankers rounding balances the
+// deviation caused by precision, especially when a large number of restakers undertake
+// undelegation. Additionally, the last undelegation from an operator will undelegate all
+// remaining token to avoid the calculated token amount is bigger than the remaining token
+// caused by the bankers rounding.
+func TokensFromShares(stakerShare, totalShare sdkmath.LegacyDec, operatorAmount sdkmath.Int) (sdkmath.Int, error) {
```

Comment:
Instead of `operatorAmount`, use `totalAmount` for clarity.

Reply:
Indeed, will rename it in the next PR.